joplin/base/signals/janis_build_triggers.py

```diff
@@ -1,131 +1,212 @@
-from base.signals.netlify_publish import netlify_publish
-from base.signals.publish_v3 import publish_v3
-from flags.state import flag_enabled
+from pages.home_page.models import HomePage
+
 
 import logging
 logger = logging.getLogger('joplin')
 
-JANIS_SLUG_URL = settings.JANIS_SLUG_URL
-
-
-def trigger_build(sender, pages_ids, action='saved', instance=None):
-    """
-    triggers different build process depending on environment
-    source = name of snippet or object triggering build
-    """
-    trigger_object = instance
-    logger.info(f'{trigger_object} {action}, triggering build')
-    if settings.IS_STAGING or settings.IS_PRODUCTION or settings.IS_REVIEW:
-        publish_v3(pages_ids)
 
-def collect_pages(instance):
+'''
+Gets data from a page that is required to send to the Publisher.
+
+returns {
+    id: Int, the page_id used by wagtail.
+    global_id: String, a hashed id used to query against graphql api
+    triggered_build: Boolean, was this the page that triggered the publish request?
+    action:
+        "published" - publish action for itself and other pages
+        "unpublished" - this page triggered the unpublish action for itself and other pages
+        "updated_by_snippet" - this page was updated by a snippet being saved or deleted
+        "saved" - a snippet is being saved, has no impact on Janis itself, but it could result in pages getting "secondary_publish_by_snippet"
+        "deleted" - a snippet is being deleted, has no impact on Janis itself, but it could result in pages getting "secondary_publish_by_snippet"
+    is_page: Boolean, is it a page as opposed to a snippet?
+    content_type: String, which specific content_type (this is more for logging)
+    author: Int, the id of the author of the latest revision
+}
+TODO: Figure out what format global_id should be in order to run queries with it.
+The current global_id is basically a placeholder.
+'''
+def get_page_data(page, triggered_build, action):
+    latest_revision = page.get_latest_revision()
+    # imported pages may not have a latest_revision yet.
+    if latest_revision:
+        author = latest_revision.user.id
+    else:
+        author = None
+    return {
+        "id": page.id,
+        "global_id": Node.to_global_id(page.content_type.name, page.id),
+        "triggered_build": triggered_build,
+        "action": action,
+        "is_page": True,
+        "content_type": page.specific_class.get_verbose_name(),
+        "author": author,
+    }
+
+
+def collect_pages(primary_page, action):
     # does this work on page deletion? pages arent deleted right, just unpublished?
     """
-    :param instance: the page that has been altered
-    :return: an array of global page ids -- note this may need to be a dictionary of page ids and type?
+    :param primary_page: the page that triggered the publish/unpublish
+    :return: pages: an array of the data from all pages impacted by the publish/unpublish event
```

Comment:
this is awesome

Reply:
It's basically not used for anything now, but it sets us up in a better place for incremental builds later.